### PR TITLE
Mam support

### PR DESCRIPTION
**Pull request by mikkelspring**
Alright, I have it working on a basic functionality level, although the XML parsing in the providers is a bit ugly. Can you review and we will discuss what else to incorporate?

### CODE REVIEW

File: src/main/scala/smack/messaging/xmmp-extensions/MAMResponse.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.joda.time.format.ISODateTimeFormat
+
+// implementation of XEP-0313 Message Archive Management, for querying XMMP server for historical messages
+// see http://xmpp.org/extensions/xep-0313.html
+
+object MAMResponse extends ExtensionInfoProvider {
+  val XmlNamespace = "urn:xmpp:mam:0"
+  val ElementName = "result"
+
+  val ForwardedNamespace = "urn:xmpp:foward:0"
```
Comment (mpollmeier) [inline src/main/scala/smack/messaging/xmmp-extensions/MAMResponse.scala]: `foward`... typo in namespace? does this work? or is that the reason why the below is commented out?

**Comment by mpollmeier** [comment]
would be good to enable/disable message archiving globally in the configuration, i.e. when globally configured, archive every message across the board.

**Comment by mpollmeier** [comment]
this repo is also the entry point for playing with the xmpp functionality, therefor it would be good to add message archiving to the command line client aswell (e.g. command `chatHistory`)

File: src/main/scala/smack/messaging/xmmp-extensions/MAMResponse.scala (hunk 1 above)
Comment (mikkelspring) [inline src/main/scala/smack/messaging/xmmp-extensions/MAMResponse.scala]: Nice catch. I got burnt out after writing the MAMRequest tests so didn't do the response yet. Will do so now.

**Comment by mikkelspring** [comment]
I got tired of the absolutely ridiculous way that smack wants you to parse the xml, so I grinded through it until I figured out the quirks and now have it building the text strings that can then be parsed directly by the element extension class object

**Comment by mpollmeier** [comment]
LGTM

**Pull request merged**